File: project-practice/kotlin-language/ClassesImplementation/src/kiz/learnwithvel/classesimplementation/javacode/Employee.java
package kiz.learnwithvel.classesimplementation.javacode;

public class Employee {

    private final String mFullName;
    private final String mDepartment;
    private final Double mSalary;
    private final String mWorkedAt;

    public Employee(String fullName, String department, Double salary) {
        mFullName = fullName;
        mDepartment = department;
        mSalary = salary;
        mWorkedAt = "Learn with Vel Inc.";
    }

    public String getFullName() {
        return mFullName;
    }

    public String getDepartment() {
        return mDepartment;
    }

    public Double getSalary() {
        return mSalary;
    }

    public String getWorkedAt() {
        return mWorkedAt;
    }
}

File: project-practice/kotlin-language/ClassesImplementation/src/kiz/learnwithvel/classesimplementation/kotlincode/Employee.kt
package kiz.learnwithvel.classesimplementation.kotlincode

import java.text.NumberFormat
import java.util.*

fun main(args: Array<String>) {

    //kotlin
    val emp = Employee("Ariel Namias Austria", "Software Developer", 100_000.00)
    println(emp.mFullName)
    println(emp.mDepartment)
    println(NumberFormat.getCurrencyInstance(Locale.US).format(emp.mSalary))
    println(emp.mWorkedAt)


    //javacode
    val emp2 = kiz.learnwithvel.classesimplementation.javacode.Employee(
        "Ariel Namias Austria",
        "Software Developer",
        100_000.00
    )
    println(emp2.fullName)
    println(emp2.department)
    println(NumberFormat.getCurrencyInstance(Locale.US).format(emp2.salary))
    println(emp2.workedAt)


}

class Employee(
    val mFullName: String,
    val mDepartment: String,
    val mSalary: Double,
    val mWorkedAt: String = "Learn with Vel Inc."
) {


}